fhir-server/src/main/java/com/ibm/fhir/server/util/FHIRRestHelper.java

```diff
@@ -2564,20 +2564,27 @@ private Bundle createSearchBundle(List<Resource> resources, FHIRSearchContext se
             }
 
             for (Resource resource : resources) {
-                if (resource.getId() == null) {
-                    throw new IllegalStateException("Returned resources must have an id.");
+                Bundle.Entry.Builder entryBuilder = Bundle.Entry.builder();
+                if (resource != null) {
+                    if (resource.getId() != null) {
+                        entryBuilder.fullUrl(Uri.of(getRequestBaseUri(type) + "/" + resource.getClass().getSimpleName() + "/" + resource.getId()));
+                    } else {
+                        String msg = "A resource with no id was found.";
+                        issues.add(FHIRUtil.buildOperationOutcomeIssue(IssueSeverity.WARNING, IssueType.NOT_SUPPORTED, msg));
```

Comment:
do these warnings make it into our operational logs?  if not, we might want to add a log.warning here too because we should really never have this case

Reply:
Added.